src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/custom.py

```diff
@@ -59,6 +61,127 @@ def get_server_location(server_name, resource_group_name):
 ###############################################
 
 
+class ClientType(Enum):
+    ado_net = 'ado.net'
+    sqlcmd = 'sqlcmd'
+    jdbc = 'jdbc'
+    php_pdo = 'php_pdo'
+    php = 'php'
+    odbc = 'odbc'
+
+class ClientAuthenticationType(Enum):
+    sql_password = 'SqlPassword'
+    active_directory_password = 'ADPassword'
+    active_directory_integrated = 'ADIntegrated'
+
+
+def _get_server_dns_suffx():
+    # Allow dns suffix to be overridden by environment variable for testing purposes
+    from os import getenv
+    return getenv('_AZURE_CLI_SQL_DNS_SUFFIX') or get_active_cloud().suffixes.sql_server_hostname
```

Comment:
The `getenv` function supports default value: https://docs.python.org/3/library/os.html?highlight=getenv#os.getenv

Reply:
Why does the env var start with underscore?
Also, should make use of the config class we have in the CLI.
https://github.com/Azure/azure-cli/blob/master/src/azure-cli-core/azure/cli/core/_config.py

e.g. https://github.com/Azure/azure-cli/blob/master/src/command_modules/azure-cli-batch/azure/cli/command_modules/batch/custom.py#L83

Reply:
This is part of the magic that allows us to use Azure CLI internally in a dev environment that doesn't have ARM frontend, so it's not really part of the officially supported surface area

Reply:
Ah okay